bundle/edu.gemini.pot/src/test/scala/edu/gemini/spModel/target/env/GuideGroupSpec.scala

```diff
@@ -177,90 +175,92 @@ class GuideGroupSpec extends Specification with ScalaCheck with Arbitraries {
+      forAll { g: GuideGroup =>
         val gpts1 = g.getAll.asScalaList
         val gpts2 = AllProbes.flatMap { gp => g.get(gp).asScalaOpt }
         gpts1 == gpts2
       }
   }
-/*
-  "GuideGroup getAll" should {
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
-
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
+  "GuideGroup putAll" should {
+    "contain the original primary guide probes and all primary guide probes for which targets have been added" in
+      forAll { (g: GuideGroup, gps: Set[GuideProbe]) =>
+        val gpts = gps.map(GuideProbeTargets.create(_, new SPTarget())).toList.asImList
+        val allGps = g.getPrimaryReferencedGuiders.asScala.toSet ++ gps
+        g.putAll(gpts).getPrimaryReferencedGuiders.asScala.toSet == allGps
       }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
-      }
   }
 
-  "GuideGroup getAll" should {
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
+  "GuideGroup setAll" should {
+    "only contain the guide probes in the collection" in
+      forAll { (g: GuideGroup, gps: Set[GuideProbe]) =>
+        val gpts = gps.map(GuideProbeTargets.create(_, new SPTarget())).toList.asImList
+        val g2 = g.setAll(gpts)
+        val gpOut = g.getReferencedGuiders.asScala.toSet -- g2.getReferencedGuiders.asScala.toSet
+        g2.getReferencedGuiders.asScala.toSet == gpts.asScalaList.map(_.getGuider).toSet &&
+          gpOut.forall(gp => !g2.contains(gp))
       }
+  }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
+  "GuideGroup getPrimaryReferencedGuiders" should {
+    "contain a guider iff it has a primary target" in
+      forAll { g: GuideGroup =>
+        g.getAll.asScalaList.collect {
+          case gpt if gpt.getPrimary.isDefined => gpt.getGuider
+        }.toSet == g.getPrimaryReferencedGuiders.asScala.toSet
       }
+  }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
+  "GuideGroup getReferencedGuiders" should {
+    "contain a guider iff it has associated targets" in
+      forAll { g: GuideGroup =>
+        g.getAll.asScalaList.map(_.getGuider).toSet == g.getReferencedGuiders.asScala.toSet
       }
+  }
 
-    "" in
-      forAll { (g: GuideGroup) =>
-        true
+  "GuideGroup removeTarget" should {
+    "remove an existing target added to all guiders and preserve the order of the original targets" in
+      forAll { (g: GuideGroup, target: SPTarget) =>
+        val oldGpts = g.getAll
+        val newGpts = g.getAll.asScalaList.map(gt => gt.update(OptionsList.UpdateOps.append(target))).asImList
+        val remGpts = g.setAll(newGpts).removeTarget(target).getAll
+        remGpts == oldGpts
```

Comment:
We should also check that removing the last target from a mapping removes the mapping altogether.  Also, removing the primary target from a "focused" `OptsList` changes the focus correctly.